### DISCUSSION TITLE
BasicViewLocatorSample: How to show Email Address which user input in second view on third view ?

### DISCUSSION THREAD

**Discussion by artwe** [AvaloniaUI/Avalonia.Samples#96] [Q&A] [answered]
Just as the title. How to share data in different user controls ?

**Comment by timunie** [comment, marked as answer]
@artwe there is no one fits all solution here. Just some thoughts:
- You can add a reference to MainViewModel and register it inside the constructor
- You can use a Messenger like this: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/messenger
- You can use DependencyInjection like this: https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection
- You can have a single static data object `public static Options MyOptions {get; }` 

And anything in between. Hope you can find one idea that suits you here.

**Reply by artwe** [reply]
Thank you!👍 I will try it later.
